tests/components/image_processing/test_init.py

```diff
@@ -7,286 +7,256 @@
 from homeassistant.core import callback
 from homeassistant.exceptions import HomeAssistantError
 from homeassistant.loader import DATA_CUSTOM_COMPONENTS
-from homeassistant.setup import setup_component
+from homeassistant.setup import async_setup_component
 
-from tests.common import (
-    assert_setup_component,
-    get_test_home_assistant,
-    get_test_instance_port,
-)
+from tests.common import assert_setup_component, get_test_instance_port
 from tests.components.image_processing import common
 
 
-class TestSetupImageProcessing:
-    """Test class for setup image processing."""
+def get_url(hass):
+    """Return camera url."""
+    state = hass.states.get("camera.demo_camera")
+    return f"{hass.config.internal_url}{state.attributes.get(ATTR_ENTITY_PICTURE)}"
 
-    def setup_method(self):
-        """Set up things to be run when tests are started."""
-        self.hass = get_test_home_assistant()
 
-    def teardown_method(self):
-        """Stop everything that was started."""
-        self.hass.stop()
+async def setup_image_processing(hass):
+    """Set up things to be run when tests are started."""
+    hass.data.pop(DATA_CUSTOM_COMPONENTS)
 
-    def test_setup_component(self):
-        """Set up demo platform on image_process component."""
-        config = {ip.DOMAIN: {"platform": "demo"}}
+    await async_setup_component(
+        hass,
+        http.DOMAIN,
+        {http.DOMAIN: {http.CONF_SERVER_PORT: get_test_instance_port()}},
+    )
 
-        with assert_setup_component(1, ip.DOMAIN):
-            setup_component(self.hass, ip.DOMAIN, config)
+    config = {ip.DOMAIN: {"platform": "test"}, "camera": {"platform": "demo"}}
 
-    def test_setup_component_with_service(self):
-        """Set up demo platform on image_process component test service."""
-        config = {ip.DOMAIN: {"platform": "demo"}}
+    await async_setup_component(hass, ip.DOMAIN, config)
+    await hass.async_block_till_done()
 
-        with assert_setup_component(1, ip.DOMAIN):
-            setup_component(self.hass, ip.DOMAIN, config)
 
-        assert self.hass.services.has_service(ip.DOMAIN, "scan")
+async def setup_image_processing_alpr(hass):
+    """Set up things to be run when tests are started."""
+    config = {ip.DOMAIN: {"platform": "demo"}, "camera": {"platform": "demo"}}
 
+    with patch(
+        "homeassistant.components.demo.image_processing."
+        "DemoImageProcessingAlpr.should_poll",
```

Comment:
Do you know why we need to patch should_poll?

Reply:
I don't know, tests pass without patching it

Reply:
These were patched already in the initial version added by https://github.com/home-assistant/core/pull/5166